gridfs_legacy.go

```diff
@@ -346,7 +346,7 @@ func (b *gridfsLegacyBucket) Pull(ctx context.Context, opts SyncOptions) error {
 
 	iterimpl, ok := iter.(*legacyGridFSIterator)
 	if !ok {
-		return errors.New("programmer error")
+		return errors.Errorf("programmer error: iterator should be legacy gridfs iterator but is actually type %T", iter)
```

Comment:
I think this should be more like:
```
errors.Errorf("programmer error: expected iterator of type %T but found type %T instead", *legacyGridFSIterator, iter)
```

Reply:
We could do this, but it would be a little awkward since I would have to construct an empty `legacyGridFS{}` struct. I instead changed the error message slightly to make it clearer.